GliaWidgets/CallVisualizer/VisitorCode/VisitorCodeView.swift

```diff
@@ -0,0 +1,172 @@
+                stackView.leadingAnchor.constraint(equalTo: leadingAnchor, constant: contentInsets.left),
+                stackView.trailingAnchor.constraint(equalTo: trailingAnchor, constant: -contentInsets.right),
+                stackView.bottomAnchor.constraint(equalTo: bottomAnchor, constant: -contentInsets.bottom),
+                stackView.topAnchor.constraint(equalTo: topAnchor, constant: contentInsets.top),
+                visitorCodeStack.heightAnchor.constraint(equalToConstant: 60),
+                closeButton.trailingAnchor.constraint(equalTo: trailingAnchor, constant: -30),
+                closeButton.bottomAnchor.constraint(equalTo: bottomAnchor, constant: -26)
+            ])
+        }
+
+        override func layoutSubviews() {
+            super.layoutSubviews()
+            setNeedsDisplay()
+
+            switch props.style.backgroundColor {
+            case .fill(let color):
+                backgroundColor = color
+            case .gradient(let colors):
+                makeGradientBackground(
+                    colors: colors,
+                    cornerRadius: cornerRadius
+                )
+            }
+            layer.shadowPath = UIBezierPath(
+                roundedRect: bounds,
+                byRoundingCorners: .allCorners,
+                cornerRadii: CGSize(width: cornerRadius, height: cornerRadius)
+            ).cgPath
+        }
+
+        func renderProps() {
+            titleLabel.font = props.style.titleFont
+            titleLabel.textColor = props.style.titleColor
+            titleLabel.text = L10n.Alert.VisitorCode.title
+            setFontScalingEnabled(
+                props.style.accessibility.isFontScalingEnabled,
+                for: titleLabel
+            )
+            renderedVisitorCode = props.visitorCode
+            poweredBy.alpha = props.isPoweredByShown ? 1 : 0
+            renderShadow()
+
+            switch props.style.closeButtonColor {
+            case .fill(let color):
+                closeButton.tintColor = color
+            case .gradient(let colors):
+                closeButton.makeGradientBackground(colors: colors)
+            }
+
+            closeButton.isHidden = props.closeButtonTap == nil
+        }
+
+        var renderedVisitorCode: String = "" {
+            didSet {
+                guard renderedVisitorCode != oldValue else { return }
+                visitorCodeStack.removeArrangedSubviews()
+                renderedVisitorCode.forEach { char in
+                    let valueLabel = NumberView().make { numberView in
+                        numberView.props = .init(character: char)
+                    }
+                    visitorCodeStack.addArrangedSubview(valueLabel)
+                }
+            }
+        }
+
+        func renderShadow() {
+            layer.shadowColor =
+            props.isShadowNeeded
+            ? UIColor.black.withAlphaComponent(0.16).cgColor
+            : UIColor.clear.cgColor
```

Comment:
looks not so readable, perhaps we can add some additional tabs?
```suggestion
            layer.shadowColor = props.isShadowNeeded
                ? UIColor.black.withAlphaComponent(0.16).cgColor
                : UIColor.clear.cgColor
```

Reply:
fixed [676c74c](https://github.com/salemove/ios-sdk-widgets/pull/448/commits/676c74ceb781899d4ca4fa8a8ab684c8a1837308)